### DISCUSSION TITLE
Fastify example

### DISCUSSION THREAD

**Discussion by kurtextrem** [nodejs/undici#1188] [Q&A] [answered]
Hi, the Fastify example seems interesting, but I'm somewhat puzzled how it should be used, since it spawns 2 servers and then fetches like this:
`Undici Client 2 -> Fastify server -> Undici Client 1 -> raw http server`
but at the same time it is mentioned 'from one backend server to another'. Does that mean one backend server would spawn both the fastify & the raw node server? But that would sound like it becomes slower than faster?

Or is it 3 different servers, one starting the request (maybe the browser so to say), then it hits the server w/ fastify, which is only a proxy for the server spawning the raw http server?

**Comment by Ethan-Arrowood** [comment, marked as answer]
The later of your explanations is the example I was going for. Essentially, request comes into Server 1 from a Browser. Server 1, using undici, then makes a subsequent request to Server 2. The response from Server 2 is then pipelined back to the Browser directly through the Pipeline streams.

**Reply by mcollina** [reply]
I would just use fastify-reply-from as it does the above.

**Reply by Ethan-Arrowood** [reply]
True, the plugin is the better way to do this, but the example is meant to show off the API functionality.
